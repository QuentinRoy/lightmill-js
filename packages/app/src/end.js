/* eslint-disable no-param-reassign */

import endTemplate from './views/end.pug';
import './views/end.scss';

export default node => {
  node.innerHTML = endTemplate();

  return {
    remove() {
      node.innerHTML = '';
    }
  };
};
