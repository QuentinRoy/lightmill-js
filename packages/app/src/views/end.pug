.lightmill-base-app.lightmill-base-app-end
  h1
    | The end
  p
    | Thank you for your participation.
