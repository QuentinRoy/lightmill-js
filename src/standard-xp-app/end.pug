div(style={"text-align": "center"})
  h1(style={"text-transform": "capitalize"})
    | The end
  p
    | Thank you for your participation.
