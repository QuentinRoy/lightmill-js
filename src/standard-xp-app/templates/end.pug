.xp-app-base.xp-app-base-end
  h1
    | The end
  p
    | Thank you for your participation.
