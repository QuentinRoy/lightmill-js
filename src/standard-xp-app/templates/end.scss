@import './common';

.#{$app-class}-end {
  text-align: center;

  h1 {
    margin-top: 2em;
    text-transform: capitalize;
    font-size: 2em;
  }
}
