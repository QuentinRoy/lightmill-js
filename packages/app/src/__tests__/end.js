/* globals document */

import end from '../end';

let node;
beforeEach(() => {
  node = document.createElement('div');
  node.innerHTML = '<div>Foo</div>';
});

describe('Waiting view', () => {
  it('can be shown', () => {
    end(node, 'message');
    expect(node).toMatchSnapshot();
  });

  it('can be removed', () => {
    const { remove } = end(node, 'message');
    remove();
    expect(node).toMatchSnapshot();
  });
});
