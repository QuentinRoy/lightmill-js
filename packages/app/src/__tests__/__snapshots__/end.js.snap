// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Waiting view can be removed 1`] = `<div />`;

exports[`Waiting view can be shown 1`] = `
<div>
  <div
    class="xp-app-base xp-app-base-end"
  >
    <h1>
      The end
    </h1>
    <p>
      Thank you for your participation.
    </p>
  </div>
</div>
`;
